pyasic/miners/backends/braiins_os.py

```diff
@@ -570,6 +573,48 @@ async def _get_uptime(self, rpc_summary: dict = None) -> Optional[int]:
             except LookupError:
                 pass
 
+    async def upgrade_firmware(self, file: Path):
+        """
+        Upgrade the firmware of the BOSMiner device.
+
+        Args:
+            file (Path): The local file path of the firmware to be uploaded.
+
+        Returns:
+            str: Confirmation message after upgrading the firmware.
+        """
+        try:
+            self.logger.info("Starting firmware upgrade process.")
+
+            if not file:
+                raise ValueError("File location must be provided for firmware upgrade.")
+
+            # Read the firmware file contents
+            async with aiofiles.open(file, "rb") as f:
+                upgrade_contents = await f.read()
+
+            # Encode the firmware contents in base64
+            encoded_contents = base64.b64encode(upgrade_contents).decode('utf-8')
```

Comment:
This seems inefficient, why not instead use scp to upload the file?

Reply:
Now, BBB boards don't have SCP.  I blame buildroot. 😛

Reply:
Well...thinking we should preferentially just send the file by writing to stdin then or something as a fallback when scp fails? No point in base64 encoding regardless.

> I blame buildroot.

Pretty sure braiinsos is OpenWRT based.